plugin/adsk/plugin/geomNode.cpp

```diff
@@ -0,0 +1,236 @@
+
+    // Inputs
+    filePathAttr
+        = typedAttrFn.create("filePath", "fp", MFnData::kString, MObject::kNullObj, &retValue);
+    typedAttrFn.setWritable(true);
+    typedAttrFn.setReadable(false);
+    CHECK_MSTATUS_AND_RETURN_IT(retValue);
+    retValue = addAttribute(filePathAttr);
+    CHECK_MSTATUS_AND_RETURN_IT(retValue);
+
+    rootPrimAttr
+        = typedAttrFn.create("rootPrim", "rp", MFnData::kString, MObject::kNullObj, &retValue);
+    typedAttrFn.setWritable(true);
+    typedAttrFn.setReadable(false);
+    CHECK_MSTATUS_AND_RETURN_IT(retValue);
+    retValue = addAttribute(rootPrimAttr);
+    CHECK_MSTATUS_AND_RETURN_IT(retValue);
+
+    // Outputs
+    outGeomAttr
+        = typedAttrFn.create("geometry", "geo", MFnData::kMesh, MObject::kNullObj, &retValue);
+    typedAttrFn.setWritable(false);
+    typedAttrFn.setReadable(true);
+    typedAttrFn.setStorable(true);
+    typedAttrFn.setArray(true);
+    typedAttrFn.setUsesArrayDataBuilder(true);
+    CHECK_MSTATUS_AND_RETURN_IT(retValue);
+    retValue = addAttribute(outGeomAttr);
+    CHECK_MSTATUS_AND_RETURN_IT(retValue);
+
+    outGeomMatrixAttr
+        = typedAttrFn.create("matrix", "tra", MFnData::kMatrix, MObject::kNullObj, &retValue);
+    typedAttrFn.setWritable(false);
+    typedAttrFn.setReadable(true);
+    typedAttrFn.setStorable(true);
+    typedAttrFn.setArray(true);
+    typedAttrFn.setUsesArrayDataBuilder(true);
+    CHECK_MSTATUS_AND_RETURN_IT(retValue);
+    retValue = addAttribute(outGeomMatrixAttr);
+    CHECK_MSTATUS_AND_RETURN_IT(retValue);
+
+    //
+    // add attribute dependencies
+    //
+    retValue = attributeAffects(filePathAttr, outGeomAttr);
+    retValue = attributeAffects(rootPrimAttr, outGeomAttr);
+    retValue = attributeAffects(filePathAttr, outGeomMatrixAttr);
+    retValue = attributeAffects(rootPrimAttr, outGeomMatrixAttr);
+    CHECK_MSTATUS_AND_RETURN_IT(retValue);
+
+    return retValue;
+}
+
+MayaUsdGeomNode::MayaUsdGeomNode()
+    : MPxNode()
+    , _cache(std::make_unique<CacheData>())
+{
+    TfRegistryManager::GetInstance().SubscribeTo<MayaUsdGeomNode>();
+}
+
+/* virtual */
+MayaUsdGeomNode::~MayaUsdGeomNode() { }
+
+/* virtual */
+MStatus MayaUsdGeomNode::compute(const MPlug& plug, MDataBlock& dataBlock)
+{
+    MStatus retValue = MS::kSuccess;
+
+    if (plug == outGeomAttr || plug == outGeomMatrixAttr) {
+        MDataHandle filePathHandle = dataBlock.inputValue(filePathAttr, &retValue);
+        CHECK_MSTATUS_AND_RETURN_IT(retValue);
+
+        MDataHandle rootPrimHandle = dataBlock.inputValue(rootPrimAttr, &retValue);
+        CHECK_MSTATUS_AND_RETURN_IT(retValue);
+
+        const std::string fileName = TfStringTrim(filePathHandle.asString().asChar());
+        const std::string rootPrim = TfStringTrim(rootPrimHandle.asString().asChar());
+
+        if (!fileName.empty()) {
+            if (_cache->loadedFile.empty() || _cache->loadedFile != fileName
+                || _cache->rootPrim != rootPrim) {
```

Comment:
Out of curiosity: will there be a mechanism to detect if the file is changed and reload, or is this under the user responsibility or is the cache considered private and should only be updated by Maya?